src/Features/Core/Portable/Shared/Utilities/PatternMatcher.cs

```diff
@@ -37,10 +37,20 @@ private struct Segment
             // respectively.
             public readonly TextChunk[] SubWordTextChunks;
 
-            public Segment(string text, bool verbatimIdentifierPrefixIsWordCharacter)
+            public Segment(string text, bool verbatimIdentifierPrefixIsWordCharacter, bool allowFuzzyMatching)
             {
-                this.TotalTextChunk = new TextChunk(text);
-                this.SubWordTextChunks = BreakPatternIntoTextChunks(text, verbatimIdentifierPrefixIsWordCharacter);
+                this.TotalTextChunk = new TextChunk(text, allowFuzzyMatching);
+                this.SubWordTextChunks = BreakPatternIntoTextChunks(
+                    text, verbatimIdentifierPrefixIsWordCharacter, allowFuzzyMatching);
+            }
+
+            public void Dispose()
```

Comment:
no.  as in we'll allocate and hte gc wil release it.  With 'dispose' we get nicer pooling so that we don't hit the GC as much.